### PR TITLE
CLOUD-1077 ds metadata checks and major refactor

### PR DESCRIPTION
**Pull request by andrewvinall**
Jira issue? CLOUD-1077
Release 6.5.0 backport required? No
6.5.0 doc changes needed? No
7.0.0 doc changes needed? No
Required README updates made? No

Apologies, the ds checks are new, but the "major refactor" is almost a re-write.  The main change is to remove the test input file (manifest.txt) and change most of the version checks from tests to a simple report.

### CODE REVIEW

File: cicd/forgeops-tests/lib/utils/am_pod.py [hunk 1]
```diff
@@ -11,88 +11,71 @@
...
-        Setup for checking commons library version
-        :param namespace The kubernetes namespace.
-        """
+        return am_metadata
 
-        logger.debug('Setting up for commons version check')
-        config_version_jar = 'config-%s.jar' % self.manifest['commons_version']
-        test_jar_filepath = os.path.join(AMPod.ROOT, 'webapps', 'am', 'WEB-INF', 'lib', config_version_jar)
-        super(AMPod, self).setup_commons_check(namespace, test_jar_filepath, AMPod.TEMP)
+    def is_expected_version(self):
```
Comment (csovant) [inline cicd/forgeops-tests/lib/utils/am_pod.py]: As we don't compare the version to an expected one maybe we should rename the method to `get_version()` or `print_version()` something like that?

File: cicd/forgeops-tests/lib/utils/am_pod.py [hunk 2]
```diff
@@ -11,88 +11,71 @@
...
-        """
-        Cleanup after checking commons library version
-        :param namespace The kubernetes namespace.
-        """
+        am_metadata = self.get_version()
+        Pod.print_table(am_metadata)
 
-        logger.debug('Cleaning up after commons version check')
-        super(AMPod, self).cleanup_commons_check(namespace, AMPod.TEMP)
+    def is_expected_commons_version(self):
```
Comment (csovant) [inline cicd/forgeops-tests/lib/utils/am_pod.py]: Same comment here to rename to `get_commons_version()` or `print_commons_version()`

File: cicd/forgeops-tests/lib/utils/am_pod.py [hunk 3]
```diff
@@ -11,88 +11,71 @@
...
-        :return: True is the commons version is as expected.
-        """
+        representative_commons_jar = AMPod.REPRESENTATIVE_COMMONS_JAR_NAME
+        lib_path = os.path.join(os.sep, 'usr', 'local', 'tomcat', 'webapps', 'am', 'WEB-INF', 'lib', )
+        super(AMPod, self).is_expected_versioned_commons_jar(lib_path, representative_commons_jar)
 
-        config_jar_properties = os.path.join('META-INF', 'maven', 'org.forgerock.commons', 'config', 'pom.properties')
-        logger.debug('Check commons version for config.jar')
-        return super(AMPod, self).is_expected_commons_version(namespace, AMPod.TEMP, config_jar_properties)
+    def is_expected_jdk(self):
```
Comment (csovant) [inline cicd/forgeops-tests/lib/utils/am_pod.py]: Same here `get_jdk_version()` or `print_jdk_version()`

File: cicd/forgeops-tests/lib/utils/am_pod.py [hunk 4]
```diff
@@ -11,88 +11,71 @@
...
+        logger.debug('Check Java version for ' + self.name)
+        return super(AMPod, self).is_expected_jdk({'openjdk version', 'openjdk version', 'openjdk version'})
 
-    def is_expected_jdk(self, namespace):
-        """
-        Return True if jdk is as expected, otherwise assert.
-        :param namespace The kubernetes namespace.
-        :return: True if jdk is as expected
-        """
+    def is_expected_os(self):
```
Comment (csovant) [inline cicd/forgeops-tests/lib/utils/am_pod.py]: Same here `get_os_version()` or `print_os_version()`

File: cicd/forgeops-tests/lib/utils/kubectl.py [hunk 5]
```diff
@@ -16,50 +15,80 @@
...
     """
     Run a kubectl exec command
-    :param command: command to run
+    :param namespace: kubernetes namespace
+    :param sub_command: command to run
     :return: (stdout, stderr) from running the command
     """
-    command = ' '.join([KUBECTL_COMMAND, '-n', namespace, 'exec', command])
+
+    assert isinstance(namespace, str), 'Invalid namespace type %s' % type(namespace)
```
Comment (csovant) [inline cicd/forgeops-tests/lib/utils/kubectl.py]: No really fan of having every time asserts at the beginning of each methods. 
Looks like more unit tests to me, we can add unit tests on our libs is needed.

File: cicd/forgeops-tests/lib/utils/pod.py [hunk 6]
```diff
@@ -42,111 +37,119 @@ def product_type(self):
...
+        assert isinstance(table_data, dict), 'Invalid title type %s' % type(table_data)
 
-        if am_manifest['jdk_version'] != amster_manifest['jdk_version']:
-            logger.warning("Different versions of java are in use.")
+        table = PrettyTable([table_data['TITLE'], table_data['DESCRIPTION']])
+        for key, value in table_data.items():
+            if key in ['TITLE', 'DESCRIPTION']:
+                continue
+            table.add_row([key, value])
+        logger.info(table)
```
Comment (markg58) [inline cicd/forgeops-tests/lib/utils/pod.py]: Need to add newline at beginning of this or the output looks like 
`2019-02-26 13:30:04 INFO +----------+-------------------------+

| openidm  | markg-openidm-openidm-0 |
+----------+-------------------------+
| VERSION  |      7.0.0-SNAPSHOT     |
| REVISION |         unknown         |
|   DATE   |      20190129151444     |
+----------+-------------------------+`

File: cicd/forgeops-tests/lib/utils/kubectl.py (hunk 5 above)
Comment (andrewvinall) [reply cicd/forgeops-tests/lib/utils/kubectl.py]: Okay, yes the are a bit messy.  However, I do think they offer a clear contract of what is expected from the calling code.  It is more of an aid to the test developer than a unit test.  I am happy to remove them and rely on Python's "duck typing"

I think adding unit tests to our libs is a great idea.

File: cicd/forgeops-tests/lib/utils/pod.py (hunk 6 above)
Comment (andrewvinall) [reply cicd/forgeops-tests/lib/utils/pod.py]: This is down to the logger output.
In time this information will be reported better elsewhere.

**Pull request merged**